servers/rendering/renderer_rd/shaders/luminance_reduce_raster.glsl

```diff
@@ -0,0 +1,65 @@
+/* clang-format off */
+#[vertex]
+
+#version 450
+
+#VERSION_DEFINES
+
+#include "luminance_reduce_raster_inc.glsl"
+
+layout(location = 0) out vec2 uv_interp;
+/* clang-format on */
+
+void main() {
+	vec2 base_arr[4] = vec2[](vec2(0.0, 0.0), vec2(0.0, 1.0), vec2(1.0, 1.0), vec2(1.0, 0.0));
+	uv_interp = base_arr[gl_VertexIndex];
+
+	gl_Position = vec4(uv_interp * 2.0 - 1.0, 0.0, 1.0);
+}
+
+/* clang-format off */
+#[fragment]
+
+#version 450
+
+#VERSION_DEFINES
+
+#include "luminance_reduce_raster_inc.glsl"
+
+layout(location = 0) in vec2 uv_interp;
+/* clang-format on */
+
+layout(set = 0, binding = 0) uniform sampler2D source_exposure;
+
+#ifdef FINAL_PASS
+layout(set = 1, binding = 0) uniform sampler2D prev_exposure;
+#endif
+
+layout(location = 0) out highp float exposure;
+
+void main() {
+	ivec2 dest_pos = ivec2(uv_interp * luminance.dest_size);
+	ivec2 src_pos = ivec2(uv_interp * luminance.source_size);
+
+	ivec2 next_pos = (dest_pos + ivec2(1)) * luminance.source_size / luminance.dest_size;
+	next_pos = max(next_pos, src_pos + ivec2(1)); //so it at least reads one pixel
+	highp vec3 source_color = vec3(0.0);
+	for (int i = src_pos.x; i < next_pos.x; i++) {
+		for (int j = src_pos.y; j < next_pos.y; j++) {
+			source_color += texelFetch(source_exposure, ivec2(i, j), 0).rgb;
+		}
+	}
+
+	source_color /= float((next_pos.x - src_pos.x) * (next_pos.y - src_pos.y));
+
+	exposure = max(source_color.r, max(source_color.g, source_color.b));
```

Comment:
Could you explain to me why this uses max? Afaik the formular for luminance should be something along the lines of r*0.21+g*0.71+b*0.07. I am guessing that the multiplication has already happened before this shader, but shouldnt it still be + instead of max?

Reply:
Hmm, maybe on the first pass that make sense. After that the luminance value is stored in the first luminance buffer and we keep running the same shader to get the average over the total image. At this point only the r channel is used.

A better solution would then be to do something like:
```
#ifdef FIRST_PASS
    exposure = r*0.21+g*0.71+b*0.07;
#else
  exposure = r
#endif
```

And again, we should make the same change in the compute shader then. @reduz do you agree?

Reply:
On chat Reduz thought the max might be slightly less realistic but it gives more consistent results. I did try with my test scene and it did tent to over saturate but that is probably because with this formula tweaks are needed in the tonemapper.

I'll leave a commented out version of that formula in case we want to revisit it.